storage/integration_test/src/integration_test.cc

```diff
@@ -386,6 +398,112 @@ const std::string kSimpleTestFile =
     "pariatur. Excepteur sint occaecat cupidatat non proident, sunt in "
     "culpa qui officia deserunt mollit anim id est laborum.";
 
+// Only test retries on desktop since Android and iOS don't have an option
+// to retry file-not-found errors and just pass-through to native
+// implementations.
+#if FIREBASE_PLATFORM_DESKTOP
+// Set a test-only flag to enable retrying 404s.
+TEST_F(FirebaseStorageTest, TestReadBytesWithMaxRetryDuration) {
+  // Enable retrying of file-not-found errors for testing.
+  bool old_value = firebase::storage::internal::g_retry_file_not_found__for_testing;
+  firebase::storage::internal::g_retry_file_not_found__for_testing = true;
+
+  int shorter_duration_sec = 2;
+  int longer_duration_sec = 6;
+  SignIn();
+
+  // Attempt to read a non-existent file, and verify that GetBytes fails.
```

Comment:
This is just a baseline test, right? We shouldn't actually check to see that it waited some time before failing?